.github/workflows/preview.yml

```diff
@@ -0,0 +1,76 @@
+name: Amplify Preview
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: gh pr checkout ${{ github.event.issue.number }}
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  conditionalone:
+    if: github.event.issue.pull_request
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: |
+          echo "is PR"
+  conditionaltwo:
+    if: github.event.comment.author_association == 'MEMBER'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "is member"
+  conditionalthree:
+    if: github.event.comment.author_association == 'CONTRIBUTOR'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "is contributor"
+  conditionalfour:
+    needs: prepare
+    if: contains(github.event.comment.body, 'generate preview')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: check if PR
+        run: echo "has comment with 'generate preview' in it"
+  deploy:
+    needs: prepare
+    if: |
+      github.event.issue.pull_request &&
+      (github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'CONTRIBUTOR') &&
+      contains(github.event.comment.body, 'generate preview')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: configure AWS credentials
+        id: creds
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_GITHUB_ACTIONS_ROLE }}
+          output-credentials: true
+      - name: set branchname env
+        id: setenvname
+        run: |
+          # use GITHUB_HEAD_REF that is set to PR source branch
+          echo "##[set-output name=setbranchname;]$(echo ${GITHUB_HEAD_REF})"
```

Comment:
It says here that we shouldn't use `set-output`, but I haven't looked into for too long - https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Reply:
Thanks for seeing it, yeah, I should be using the `GITHUB_OUTPUT` env var instead.